manticore/core/executor.py

```diff
@@ -69,23 +46,32 @@ def newFunction(self, *args, **kw):
             self._lock.release()
     return newFunction
 
-class ProfilingResults(object):
-    def __init__(self, raw_stats, instructions_executed):
-        self.raw_stats = raw_stats
-        self.instructions_executed = instructions_executed
 
-        self.time_elapsed = raw_stats.total_tt
+class Policy(object):
+    ''' Base class for prioritization of state search '''
+    def __init__(self):
+        pass
+
+    def features(self, state):
+        ''' Save state features for prioritization before a state is stored '''
+        pass
+
+    def priority(self, state_id):
+        ''' A numeric value representing likelihood to reach the interesting program spot '''
+        return 1.0
+
+class Random(Policy):
+    def __init__(self):
+        super(Random, self).__init__()
+
+    def features(self, state):
+        ''' Save state features for prioritization before a state is stored '''
+        pass
+
+    def priority(self, state_id):
+        ''' A numeric value representing likelihood to reach the interesting program spot '''
+        return 1.0
 
-        self.loading_time = 0
-        self.saving_time = 0
-        self.solver_time = 0
-        for (func_file, _, func_name), (_, _, _, func_time, _) in raw_stats.stats.iteritems():
-            if func_name == '_getState':
-                self.loading_time += func_time
-            elif func_name == '_putState':
-                self.saving_time += func_time
-            elif func_file.endswith('solver.py') and 'setstate' not in func_name and 'getstate' not in func_name and 'ckl' not in func_name:
-                self.solver_time += func_time
 
     def __str__(self):
```

Comment:
Let's discuss this in a new PR? Since master is no longer stable, imo it's fine if we merge some slightly buggy profiling code, and make a new PR to refactor and fix .